packages/lit-html/src/lit-html.ts

```diff
@@ -774,11 +871,29 @@ export class NodePart {
         ? node.nextSibling === null
         : node === this._endNode.previousSibling)
     ) {
+      if (ENABLE_EXTRA_SECURITY_HOOKS) {
+        if (this._textSanitizer === undefined) {
+          this._textSanitizer = sanitizerFactory(node, 'data', 'property');
+        }
+        value = this._textSanitizer(value);
+      }
       // If we only have a single text node between the markers, we can just
       // set its value, rather than replacing it.
       (node as Text).data = value as string;
     } else {
-      this._commitNode(d.createTextNode(value as string));
+      const textNode = document.createTextNode('');
+      this._commitNode(textNode);
+      if (ENABLE_EXTRA_SECURITY_HOOKS) {
+        // When setting text content, for security purposes it matters a lot
+        // what the parent is. For example, <style> and <script> need to be
+        // handled with care, while <span> does not. So first we need to put a
+        // text node into the document, then we can sanitize its contentx.
+        if (this._textSanitizer === undefined) {
+          this._textSanitizer = sanitizerFactory(textNode, 'data', 'property');
+        }
+        value = this._textSanitizer(value);
+      }
+      textNode.data = value as string;
```

Comment:
In the case of `<style>` and `<script>` this might be a detached Text because of the check in `_commitNode`, yes? Can you add a comment for that?

Reply:
No, the check in `_commitNode` calls `_insert` which writes immediately to `this.startNode`.